includes/rest-api/class-sensei-rest-api-course-users-controller.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * File contains the Sensei_REST_API_Course_Users_Controller class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Course users actions controller class.
+ *
+ * @since x.x.x
+ */
+class Sensei_REST_API_Course_Users_Controller extends \WP_REST_Controller {
+
+	/**
+	 * Routes namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Routes prefix.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'course-users/batch';
+
+	/**
+	 * Sensei_REST_API_Course_Structure_Controller constructor.
+	 *
+	 * @param string $namespace Routes namespace.
+	 */
+	public function __construct( $namespace ) {
+		$this->namespace = $namespace;
+	}
+
+	/**
+	 * Register the REST API endpoints for Course Structure.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'add_users_to_courses' ],
+					'permission_callback' => [ $this, 'can_add_users_to_courses' ],
```

Comment:
I guess the same naming convention could be applied for the premissions callback. Maybe something like `batch_items_permissions_check` or `batch_(create|delete)_items_permissions_check` if the permission check for create/delete is different.